libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -3199,7 +3199,7 @@ void WaveTrack::SplitAt(double t)
 {
    for (const auto &c : mClips)
    {
-      if (c->WithinPlayRegion(t))
+      if (c->SplitsPlayRegion(t))
```

Comment:
This simple change raises similar doubts.  But I presume you have retested the destructive effects with this change?  This PR is based on all of that other stuff.

I have in fact noticed zero-length clips sometimes as a thing that ought to be debugged later.

Is there a way of making zero length clips in 3.3.3 that gets fixed by this?

Maybe there are existing issues about them?

Reply:
Not that I know of. #2263 wrongly reports zero-length clips, but that's actually the click-to-merge vertical line.
In any case, it'd be wrong indeed to split a track on the first sample of a clip.
One could even question whether a track should be splittable at the second-last sample of a clip, leaving a length-1 clip, barely more usable than a length-0 clip. But I didn't venture that far.